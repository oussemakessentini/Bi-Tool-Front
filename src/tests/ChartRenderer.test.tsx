import React from 'react';
import { render , screen, waitFor } from '@testing-library/react';
import ChartRenderer from '../components/chartRenderer';
import cubejs, { ChartType } from '@cubejs-client/core';
import { CubeProvider } from '@cubejs-client/react';

// Mock the echarts library
jest.mock('echarts', () => ({
  init: jest.fn(),
  dispose: jest.fn(),
}));
const chartType : ChartType = "line"
const cubejsApi = cubejs({
    apiUrl: 'http://localhost:4000/cubejs-api/v1',
  });
describe('ChartRenderer component', () => {
  test('renders chart component based on vizState', async () => {
    // Mock the props
    const vizState = {
      query: { measures: ["Users.count"] },
      chartType: chartType,
    };
    const chartHeight = '400px';

    const { container } = render(
        <CubeProvider cubejsApi={cubejsApi}><ChartRenderer vizState={vizState} chartHeight={chartHeight} /></CubeProvider>
      
    );

    await waitFor(()=>{
        expect(container).toMatchSnapshot();
    })

  });
});
