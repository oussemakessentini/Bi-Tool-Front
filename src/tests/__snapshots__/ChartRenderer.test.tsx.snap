// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartRenderer component renders chart component based on vizState 1`] = `
<div>
  <div
    class="sc-bgqQcB iJsTRS"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning css-dev-only-do-not-override-w8mnev"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
